logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/auth/LoginUseCase.kt

```diff
@@ -24,7 +24,8 @@ interface LoginUseCase {
         userIdentifier: String,
         password: String,
         shouldPersistClient: Boolean,
-        serverConfig: ServerConfig
+        serverConfig: ServerConfig,
+        shouldStoreSession: suspend (AuthSession) -> Boolean = { true }
```

Comment:
Why a lambda?

Do we have a need to solve this as it goes in Reloaded?
What kind of criteria would one check here?

I wonder if it doesn't lead to some spaghettification. The consumers of this will have to worry about the session persistence multiple times.

```kotlin
val authResult = loginUsingEmail(handle, password) { authSession ->
    authSession.userId == "lorem Ipsum"
}

if(authResult is AuthenticationResult.Success){
    // We don't know what happened until we check again.
    val isSessionPersisted = authResult.userSession.userId == "lorem Ipsum"
}
```

----

### Another Approach

I think that if you need a conditional storing or not of a session, you could just do:


```kotlin
// DO NOT persist by default
val authResult = loginUsingEmail(handle, password, shouldPersistSession = false)

// Only persist after checking
if(authResult is AuthenticationResult.Success){
    val shouldPersistSession = authResult.userSession.userId == "lorem Ipsum"
    if(shouldPersistSession) {
         saveSessionUseCase(authResult.userSession)
    }
}
```

Reply:
I wanted it to be simple, but actually it may be more difficult to understand and prepared specifically for Reloaded. You made me realize that we already have separate usecases for this. 😅 So I turned it into a regular Boolean.